cfme/cloud/flavor.py

```diff
@@ -108,11 +143,71 @@ class Flavor(BaseEntity, Taggable):
     name = attr.ib()
     provider = attr.ib()
 
+    def delete(self, wait=True, cancel=False):
+        """Delete current falvor """
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove Flavor', handle_alert=not cancel)
+        view.flash.assert_success_message(
+            'Delete of Flavor "{}" was successfully initiated.'.format(self.name))
+
+        if wait:
+            """Wait for flavor to disappear"""
+            wait_for(lambda: not self.exists,
+                     timeout=600,
+                     message='Wait for flavor to disappear',
+                     delay=20,
+                     fail_func=self.refresh)
+
+    def refresh(self):
+        """Refresh provider relationships and browser"""
+        self.provider.refresh_provider_relationships()
+        self.browser.refresh()
+
+    @property
+    def exists(self):
+        try:
+            navigate_to(self, 'Details')
+            return True
+        except FlavorNotFound:
+            return False
+
 
 @attr.s
 class FlavorCollection(BaseCollection):
     ENTITY = Flavor
 
+    def create(self, name, provider, ram, vcpus, disk, swap, rxtx, is_public=True, cancel=False,
+            wait=True):
+        """Create new falvor"""
+        view = navigate_to(self, 'Add')
+        form_params = {'provider': provider.name,
+                       'flavor_name': name,
+                       'ram_size': ram,
+                       'vcpus': vcpus,
+                       'disk_size': disk,
+                       'swap_size': swap,
+                       'rxtx_factor': rxtx,
+                       'public': is_public}
+
+        view.fill(form_params)
+
+        if cancel:
+            view.cancel.click()
+            flash_message = "Add of Flavor cancelled by user."
+        else:
+            view.add.click()
+            flash_message = 'Add of Flavor "{}" was successfully initialized.'.format(name)
+
+        view = self.create_view(FlavorAllView)
+        view.flash.assert_message(flash_message)
+        flavor = self.instantiate(name, provider)
+
+        if wait:
+            wait_for(lambda: flavor.exists, delay=50, timeout=600, fail_func=flavor.refresh,
```

Comment:
Though this page performance is poor, I think this delay needs to be lowered as much as possible. Is there a reason for it being as high as 50s? Can it be called every 5 seconds instead? We don't want to wait 50s if we could have waited 15